### ISSUE TITLE
`arithmetic` is not an allowed trait for strings on Windows

### ISSUE THREAD

**Issue by DCPRevere** [doom/strong_type#2]
Adding the `arithmetic` trait for a string base type so that it can be used as a key in a map results in a build failure in Windows, but succeeds in Linux.

`'st::traits::operator ==' cannot result in a constant expression`

Windows: `MSVC 19.16.27032.1`
Linux: `gcc (Ubuntu 8.3.0-6ubuntu1) 8.3.0`

```
C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/traits.hpp(112): error C3615: constexpr function 'st::traits::operator ==' cannot result in a constant expression (compiling source file C:\projects\antara-makerbot\src\order_manager\order.manager.cpp) [C:\projects\antara-makerbot\cmake-build-debug\src\mmbot-test.vcxproj]
  C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/traits.hpp(114): note: failure was caused by call of undefined function or one not declared 'constexpr' (compiling source file C:\projects\antara-makerbot\src\order_manager\order.manager.cpp)
  C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/traits.hpp(114): note: see usage of 'std::operator ==' (compiling source file C:\projects\antara-makerbot\src\order_manager\order.manager.cpp)
  C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/traits.hpp(113): note: while compiling class template member function 'bool st::traits::operator ==(const T &,const OtherOperandT &) noexcept'
          with
          [
              T=st::type<std::string,antara::order_id_tag,st::arithmetic>,
              OtherOperandT=st::type<std::string,antara::order_id_tag,st::arithmetic>
          ] (compiling source file C:\projects\antara-makerbot\src\order_manager\order.manager.cpp)
  C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/traits.hpp(197): note: see reference to class template instantiation 'st::traits::equality_comparable<T,T>' being compiled
          with
          [
              T=st::type<std::string,antara::order_id_tag,st::arithmetic>
          ] (compiling source file C:\projects\antara-makerbot\src\order_manager\order.manager.cpp)
  C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/type.hpp(64): note: see reference to class template instantiation 'st::traits::arithmetic<st::type<std::string,antara::order_id_tag,st::arithmetic>>' being compiled (compiling source file C:\projects\antara-makerbot\src\order_manager\order.manager.cpp)
  C:\projects\antara-makerbot\src\utils/mmbot_strong_types.hpp(129): note: see reference to class template instantiation 'st::type<std::string,antara::order_id_tag,st::arithmetic>' being compiled (compiling source file C:\projects\antara-makerbot\src\order_manager\order.manager.cpp)
  orders.cpp
C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/traits.hpp(112): error C3615: constexpr function 'st::traits::operator ==' cannot result in a constant expression (compiling source file C:\projects\antara-makerbot\src\orders\orders.cpp) [C:\projects\antara-makerbot\cmake-build-debug\src\mmbot-test.vcxproj]
  C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/traits.hpp(114): note: failure was caused by call of undefined function or one not declared 'constexpr' (compiling source file C:\projects\antara-makerbot\src\orders\orders.cpp)
  C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/traits.hpp(114): note: see usage of 'absl::operator ==' (compiling source file C:\projects\antara-makerbot\src\orders\orders.cpp)
  C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/traits.hpp(113): note: while compiling class template member function 'bool st::traits::operator ==(const T &,const OtherOperandT &) noexcept'
          with
          [
              T=st::type<absl::uint128,antara::price_tag,st::arithmetic>,
              OtherOperandT=st::type<absl::uint128,antara::price_tag,st::arithmetic>
          ] (compiling source file C:\projects\antara-makerbot\src\orders\orders.cpp)
  C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/traits.hpp(197): note: see reference to class template instantiation 'st::traits::equality_comparable<T,T>' being compiled
          with
          [
              T=st::type<absl::uint128,antara::price_tag,st::arithmetic>
          ] (compiling source file C:\projects\antara-makerbot\src\orders\orders.cpp)
  C:\projects\antara-makerbot\cmake-build-debug\_deps\doom_st-src\include\st/type.hpp(64): note: see reference to class template instantiation 'st::traits::arithmetic<st::type<absl::uint128,antara::price_tag,st::arithmetic>>' being compiled (compiling source file C:\projects\antara-makerbot\src\orders\orders.cpp)
```

**Comment by doom** [comment]
Hello,

This bug can be reproduced with MSVC up to `19.21`, but seems to be fixed by version `19.22`.
See [this godbolt link](https://godbolt.org/z/SZ7I-T), where it works with the latest versions of the three major compilers.

**Comment by doom** [comment]
Closing, as this was an issue with MSVC rather than with the code itself.
